cluster/kubernetes/kubernetes.go

```diff
@@ -281,7 +282,7 @@ func (c *Cluster) PublicSSHKey(regenerate bool) (ssh.PublicKey, error) {
 // the Flux instance is expected to have access to and can look for resources inside of.
 // It returns a list of all namespaces unless an explicit list of allowed namespaces
 // has been set on the Cluster instance.
-func (c *Cluster) getAllowedAndExistingNamespaces() ([]apiv1.Namespace, error) {
+func (c *Cluster) getAllowedAndExistingNamespaces(ctx context.Context) ([]apiv1.Namespace, error) {
```

Comment:
There's not much point in passing the context down to the leaves unless it's either given to the remote calls, or at least consulted. Since client-go doesn't support contexts, the only choice is the latter: in other words, to check that the context hasn't closed before making remote calls. It may time out or be cancelled _during_ the operation (including the time spent waiting for the rate limiter), but at least further operations won't be attempted, so overall, pressure will be lower.

The easiest way to check the context in this circumstance will be to see if `ctx.Err()` is `nil` or not:

```
if err := ctx.Err(); err != nil {
  return nil, err
}
ns, err := c.client.CoreV1().Namespaces().Get(name, meta_v1.GetOptions{})
// ...
```